Test/AbstractBlockServiceTestCase.php

```diff
@@ -97,4 +97,20 @@ protected function assertSettings(array $expected, BlockContextInterface $blockC
 
         $this->assertSame($completeExpectedOptions, $blockSettings);
     }
+
+    /**
+     * NEXT_MAJOR: Remove this method when dropping support for < PHPUnit 5.4.
+     *
+     * @param string $class
+     *
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    protected function createMock($class)
```

Comment:
> we will still have to change the `extends`

Maybe, but it's just a line.

If we don't create a base class, in this case, it would be the same to add a condition to each `createMock` calls instead.

BTW, this is the way we are doing for SF BC, why not here?